### PR TITLE
Add minimal mypy type checking

### PR DESCRIPTION
**Pull request by tomwhite**

### CODE REVIEW

**Comment by tomwhite** [comment]
This is passing now (except for a spurious code coverage failure). Ideally we would run `mypy` with `--strict`, but this is a good start.

**Comment by TomAugspurger** [comment]
Merged in #50.

For now I'd prefer to not run `mypy` in pre-commit. I haven't had good experience with mypy running in diff mode (lots of incorrect typing errors).

cc @shoyer if you have any thoughts on adding mypy to a project.

File: setup.py [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "flake8",
     "black",
     "codecov",
+    "mypy",
```
Comment (shoyer) [inline setup.py:35]: We should pin the mypy version for rechunker's CI:
https://mypy.readthedocs.io/en/stable/existing_code.html#continuous-integration

**Review by shoyer — APPROVED** [review]
Thanks for these fixes! This looks great.

**Comment by TomAugspurger** [comment]
Thanks!

**Pull request merged**